web-common/src/features/dashboards/rows-viewer-accordion/RowsViewerAccordion.svelte

```diff
@@ -0,0 +1,113 @@
+<script lang="ts">
+  import CaretUpIcon from "@rilldata/web-common/components/icons/CaretUpIcon.svelte";
+  import RowsViewer from "../rows-viewer/RowsViewer.svelte";
+  import CaretDownIcon from "@rilldata/web-common/components/icons/CaretDownIcon.svelte";
+  import { createQueryServiceMetricsViewTotals } from "@rilldata/web-common/runtime-client";
+  import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+  import { formatCompactInteger } from "@rilldata/web-common/lib/formatters";
+  import { useDashboardStore } from "../dashboard-stores";
+  import { useModelHasTimeSeries } from "../selectors";
+  export let metricViewName: string;
+  let isOpen = false;
+  const toggle = () => {
+    isOpen = !isOpen;
+  };
+
+  $: totalsQuery = createQueryServiceMetricsViewTotals(
+    $runtime.instanceId,
+    metricViewName,
+    {
+      measureNames: ["count"],
+      inlineMeasures: [
+        {
+          name: "count",
+          expression: "count(*)",
+        },
+      ],
+    },
+    {
+      query: {
+        queryKey: ["dashboardAllRowsCt"],
+        enabled: true,
+      },
+    }
+  );
+
+  $: dashboardStore = useDashboardStore(metricViewName);
+
+  $: metricTimeSeries = useModelHasTimeSeries(
+    $runtime.instanceId,
+    metricViewName
+  );
+  $: hasTimeSeries = $metricTimeSeries.data;
+
+  $: timeStart = $dashboardStore.selectedTimeRange?.start?.toISOString();
+  let timeEnd;
+  $: {
+    let maybeEnd = $dashboardStore.selectedTimeRange?.end;
+    if (maybeEnd) {
+      timeEnd = new Date(maybeEnd.valueOf() + 1).toISOString();
+    }
+  }
+
+  $: filteredTotalsQuery = createQueryServiceMetricsViewTotals(
+    $runtime.instanceId,
+    metricViewName,
+    {
+      measureNames: ["count"],
+      inlineMeasures: [
+        {
+          name: "count",
+          expression: "count(*)",
+        },
+      ],
+      timeStart: timeStart,
+      timeEnd: timeEnd,
+      filter: $dashboardStore?.filters,
+    },
+    {
+      query: {
+        queryKey: [
+          "dashboardFilteredRowsCt",
+          {
+            timeStart,
+            timeEnd,
+            filter: $dashboardStore?.filters,
+          },
+        ],
+        enabled:
+          (hasTimeSeries ? !!timeStart && !!timeEnd : true) &&
+          !!$dashboardStore?.filters,
+      },
+    }
+  );
+
+  let label = "";
+
+  $: {
+    if ($totalsQuery.data && $filteredTotalsQuery.data) {
+      label = `${formatCompactInteger(
+        $filteredTotalsQuery.data.data.count
+      )} of ${formatCompactInteger($totalsQuery.data.data.count)} rows`;
+    }
+  }
+</script>
+
+<div>
+  <button
+    class="w-full bg-gray-100 h-7 text-left px-2 border-t border-t-gray-200 text-xs text-gray-800 flex items-center gap-1"
+    on:click={toggle}
+    aria-label="Toggle rows viewer"
+  >
+    <span class="font-bold">Source Data</span>
+    {label}
+    {#if isOpen}
+      <CaretUpIcon size="14px" />
+    {:else}
+      <CaretDownIcon size="14px" />
+    {/if}
```

Comment:
Nit: elsewhere we're using an animation like this to flip between up/down:

```svelte
<span class="{active ? '-rotate-180' : ''} transition-transform">
    <CaretDownIcon />
</span>
```

Reply:
I tried it but it has no UX affect because you can't eyes can't see the animation happening while the icon is moving to a different location on screen. So won't do